### PR TITLE
feat: add go.consumer.close.timeout.ms to bound Close() duration

### PR DESCRIPTION
**Pull request by nolansherman-spoton**
What
----

Consumer.Close() can hang indefinitely when the Kafka broker is unresponsive during the leave-group handshake (particularly with static membership). This adds an opt-in go.consumer.close.timeout.ms config property that bounds the rd_kafka_consumer_closed polling loop.

When the timeout expires, Close() calls rd_kafka_destroy_flags with RD_KAFKA_DESTROY_F_NO_CONSUMER_CLOSE to force-free all resources without waiting for broker acknowledgement.

Default value is 0 (no timeout), preserving existing behaviour. No breaking changes.

Resolves hangs reported in:
- #767 (Consumer hanging on Close() when redeploying)
- #1016 (rd_kafka_rebalance_protocol() hang during close)

Checklist
------------------
- [x] Contains customer facing changes? Including API/behavior changes 
- [x] Did you add sufficient unit test and/or integration test coverage for this PR?
  - If not, please explain why it is not required

Tradeoffs
----------

When the timeout fires:
- The consumer does not perform a clean LeaveGroup — for static members, the broker retains the assignment until `session.timeout.ms` expires (identical to a pod crash)
- Any uncommitted offsets since the last auto-commit are lost (also identical to a crash)

These are acceptable tradeoffs vs. an indefinite hang that causes OOM kills.

Test & Review
------------

```bash
# Requires Docker for the Kafka broker container
go test -v -timeout 5m -run TestConsumerCloseTimeout ./kafka/
```

- `TestConsumerCloseTimeout_HangsWithoutConfig`: pauses the broker and confirms Close() hangs >45s without the config
- `TestConsumerCloseTimeout_CompletesWithConfig`: same scenario but with `go.consumer.close.timeout.ms=5000`, confirms Close() returns in ~5-10s

Open questions / Follow-ups
--------------------------

### CODE REVIEW

File: kafka/consumer_close_timeout_test.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	done := make(chan error, 1)
+	go func() { done <- consumer.Close() }()
+
+	select {
+	case <-done:
+		t.Log("Close() returned -- hang not reproduced in this run")
+	case <-time.After(45 * time.Second):
+		t.Log("CONFIRMED: Close() hung for >45s without go.consumer.close.timeout.ms")
+		t.Log("The consumer goroutine is now permanently leaked with all C-heap resources.")
+	}
```
[the remark this answers is not in the document]
Comment (JacekPietrasSpotOn) [reply kafka/consumer_close_timeout_test.go:169]: I'm checking it heavely with AI that explain it to me and it seems valid

File: kafka/consumer_close_timeout_test.go [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+// Requirements: Docker (for testcontainers Kafka broker)
+//
+// Run: go test -v -timeout 5m -run TestConsumerCloseTimeout ./kafka/
+
+// startBroker starts a Kafka container and returns the bootstrap address.
+func startBroker(t *testing.T) (bootstraps string, containerID string, cleanup func()) {
+	t.Helper()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	cmd := exec.CommandContext(ctx, "docker", "run", "-d", "--rm",
```
[the remark this answers is not in the document]
Comment (JacekPietrasSpotOn) [reply kafka/consumer_close_timeout_test.go:25]: I'm checking it heavely with AI that explain it to me and it seems valid

File: kafka/consumer.go [hunk 3]
```diff
@@ -628,6 +664,12 @@ func NewConsumer(conf *ConfigMap) (*Consumer, error) {
 	}
 	eventsChanSize := v.(int)
 
+	v, err = confCopy.extract("go.consumer.close.timeout.ms", 0)
+	if err != nil {
+		return nil, err
+	}
+	c.closeTimeoutMs = v.(int)
```
[the remark this answers is not in the document]
Comment (JacekPietrasSpotOn) [reply kafka/consumer.go:671]: negative timeout is a edge case, I would ignore

File: kafka/consumer_close_timeout_test.go [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+// These tests demonstrate that Consumer.Close() hangs indefinitely when the
+// broker is unresponsive during static member leave-group, and that configuring
+// go.consumer.close.timeout.ms resolves the issue by force-destroying the
+// consumer after the timeout.
+//
+// Requirements: Docker (for testcontainers Kafka broker)
+//
+// Run: go test -v -timeout 5m -run TestConsumerCloseTimeout ./kafka/
+
+// startBroker starts a Kafka container and returns the bootstrap address.
```
[the remark this answers is not in the document]
Comment (JacekPietrasSpotOn) [reply kafka/consumer_close_timeout_test.go:20]: It's just a consistency issue. It's ok to skip

File: kafka/consumer.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (nolansherman-spoton) [reply kafka/consumer.go:671]: Yeah, why would a user expect anything meaningful from a negative timeout.

File: kafka/consumer_close_timeout_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (nolansherman-spoton) [reply kafka/consumer_close_timeout_test.go:20]: Yes, I am happy to cleanup if the confluent team wants, but I think this demonstrates the issue well enough.

**Comment by nolansherman-spoton** [comment]
I am closing this PR in favor of a more targeted approach in the following PR: https://github.com/confluentinc/confluent-kafka-go/pull/1588
